libs/hooks/src/store/use-akasha-store.ts

```diff
@@ -1,11 +1,17 @@
-import { AkashaProfile, IUserStore } from '@akashaorg/typings/lib/ui';
 import { useRootComponentProps } from '../use-root-props';
 import { useSyncExternalStore } from 'react';
+import { getProfileInfo } from './get-profile-info';
+import { AuthenticationStore } from './authentication-store';
+import { AkashaProfile } from '@akashaorg/typings/lib/ui';
 
 export function useAkashaStore() {
   const { plugins } = useRootComponentProps();
-  const userStore = plugins['@akashaorg/app-profile']?.profile
-    ?.userStore as IUserStore<AkashaProfile>;
-  const data = useSyncExternalStore(userStore?.subscribe, userStore?.getSnapshot);
-  return { data, userStore };
+  const authenticationStore = AuthenticationStore.getInstance<AkashaProfile>(
+    plugins['@akashaorg/app-profile']?.profile?.getProfileInfo || getProfileInfo,
```

Comment:
I would rename the fallback `getProfileInfo` to `getAkashaProfileInfo`. will make it more clear

Reply:
I'm still not very happy that the authenticationStore works only with `plugins['@akashaorg/app-profile']` plugin. I'm thinking more and more on splitting this into 2 stores without any deps between them. A store that requires a DID and implements a profile plugin. Can you create an issue with this? We can work on it afterwards

Reply:
You mean one store just for getting the plugin that has function to fetch profile info and the other one the authentication store ?

Reply:
yes, well a store that implements only one method does not make much sense now, right? This is why it's not a high priority right now. But in the near future we can  include the methods to handle all CRUD operations via the profile plugin.

Reply:
done, I have created the issue on linear